### PR TITLE
Fix PHP 5.2 support

### LINKED ISSUE #441: Lost support for PHP 5.2
Since now are using closures, Storefront now supports only PHP 5.3 or later...

https://github.com/woothemes/storefront/blob/master/inc/class-storefront.php#L339-L341

https://github.com/woothemes/storefront/blob/master/inc/customizer/class-storefront-customizer.php#L74-L76

### PR DESCRIPTION
**Pull request by claudiosmweb**
Closes #441

### CODE REVIEW

File: inc/class-storefront.php [hunk 1]
```diff
@@ -336,12 +336,33 @@ public function get_structured_data() {
 			} else {
 				$structured_data = $structured_data + self::$structured_data[0];
 			}
-      array_walk_recursive( $structured_data, function ( &$value ) {
-        $value = sanitize_text_field( $value );
-      } );
```
Comment (opportus) [inline inc/class-storefront.php]: Oops, sorry.

File: inc/class-storefront.php [hunk 2]
```diff
@@ -336,12 +336,33 @@ public function get_structured_data() {
...
+					$sanitized_value = $this->sanitize_structured_data( $value );
+				} else {
+					$sanitized_value = sanitize_text_field( $value );
+				}
+
+				$sanitized[ sanitize_text_field( $key ) ] = $sanitized_value;
+			}
+
+			return $sanitized;
+		}
```
Comment (opportus) [inline inc/class-storefront.php]: Does `array_walk_recursive()` not recommended to use here ?

Comment (opportus) [inline inc/class-storefront.php]: @claudiosmweb 

Does `array_walk_recursive()` not recommended to use here ?

Comment (opportus) [inline inc/class-storefront.php]: @claudiosmweb 

Does `array_walk_recursive()` not good to use here ?

Comment (claudiosmweb) [inline inc/class-storefront.php]: Does `array_walk_recursive()` sanitize the array keys too? :P

Comment (claudiosmweb) [inline inc/class-storefront.php]: Only for values we can use only `wc_clean()`, but since I want to sanitize the keys too, I made this method.

Comment (opportus) [inline inc/class-storefront.php]: I forgot that, I'll update this on my WooCommerce PR.

Comment (opportus) [inline inc/class-storefront.php]: I forgot that I had to sanitize keys, thanks...

Comment (opportus) [inline inc/class-storefront.php]: @claudiosmweb 
I forgot that I had to sanitize keys...
Also, you just learn me something pretty usefull with the method/function calling itself which by this way, recursively walk through the array, thanks...

**Pull request merged**